src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -374,11 +397,43 @@ async def _update_repo_using_branch_head(
     return f"{repo.repo_id}:{repo.branch}:{sha}"
 
 
-async def _check_repositories(repos: list[GitRepo]) -> dict:
+async def _check_repositories(
+    repos: list[GitRepo], syncedViaTags: bool = False
+) -> dict:
     changes = {}
     for repo in repos:
         log.debug("fetching repo: %s...", repo.repo_url)
         await _git_fetch(repo.directory)
+    latestTags = [
+        {
+            repo.repo_id: await _git_get_latest_matching_tag_capture_groups(
+                repo.directory, repo.tags
+            )
+        }
+        for repo in repos
+    ]
+    uniqueLatestTags = list(
+        {
+            list(i.values())[0][0] if list(i.values())[0] else None
```

Comment:
here is pretty cryptic to read what you intend to do

TIP: 
- structure first the output of ``await _git_get_latest_matching_tag_capture_groups`` e.g. using something like a ``TagInfo`` dataclass. Then the code is not only more readable but much more maintainable

Reply:
this is true but for the sake of moving on for now (please excuse this) I will leave as-is for now :/